server/storage.lua

```diff
@@ -54,10 +54,16 @@ function FetchVehicleEntityByPlate(plate)
     return MySQL.single.await('SELECT * FROM player_vehicles WHERE plate = ?', {plate})
 end
 
----@param plate string
+---@param vehicleId integer
+---@return VehicleEntity
+function FetchVehicleEntityByVehicleId(vehicleId)
+    return MySQL.single.await('SELECT * FROM player_vehicles WHERE vehicleId = ?', {vehicleId})
+end
+
+---@param vehicle integer|string
 ---@return boolean
-function DoesVehicleEntityExist(plate)
-    local count = MySQL.scalar.await('SELECT COUNT(*) FROM player_vehicles WHERE plate = ?', {plate})
+function DoesVehicleEntityExist(vehicle)
+    local count = MySQL.scalar.await('SELECT COUNT(*) FROM player_vehicles WHERE vehicleId = ? or plate = ?', {vehicle, vehicle})
```

Comment:
What about the caller? Doesn't the parameter need to be vehicleId? Passing in the vehicle would be the entityId. vehicleId is a auto incremented column on the player_vehicles table and can be gotten using 

Entity(vehicle).state.vehicleid or exports.qbx_vehicles:GetVehicleIdByPlate(GetVehicleNumberPlateText(vehicle))

Reply:
It is used in https://github.com/Qbox-project/qbx_vehicleshop/blob/f75a0090b87928406a1e86cb7f0a516fa093944a/server/main.lua#L85 which I wasn't going to edit (maybe I should) and there is plate, so I tried to handle both cases with one input variable

Reply:
DoesVehicleEntityExist is used for generating a unique plate when buying a vehicle. However, https://docs.qbox.re/resources/qbx_vehicles/exports/server#createplayervehicle already guarantees a unique plate, so the calling function can be removed.